### PR TITLE
Migrate fetch of surveys from local db to Coroutines Flow

### PR DESCRIPTION
**Pull request by gino-m**
First step towards migrating local db layer to Coroutines (#1581).

@JSunde PTAL?

### CODE REVIEW

File: ground/src/main/java/com/google/android/ground/persistence/local/LocalDatabaseModule.kt [hunk 1]
```diff
@@ -18,22 +18,30 @@ package com.google.android.ground.persistence.local
...
+  fun localDatabase(
+    @ApplicationContext context: Context,
+    @IoDispatcher ioDispatcher: CoroutineDispatcher
+  ): LocalDatabase {
     return Room.databaseBuilder(context, LocalDatabase::class.java, Config.DB_NAME)
       .fallbackToDestructiveMigration() // TODO(#128): Disable before official release.
+      // Run queries and transactions on background I/O thread.
+      .setQueryExecutor(ioDispatcher.asExecutor())
       .build()
   }
```
Comment (JSunde) [inline ground/src/main/java/com/google/android/ground/persistence/local/LocalDatabaseModule.kt:46]: nit: this could be a single-expression function

**Review by JSunde — APPROVED** [review]

File: ground/src/main/java/com/google/android/ground/persistence/local/LocalDatabaseModule.kt (hunk 1 above)
Comment (gino-m) [reply ground/src/main/java/com/google/android/ground/persistence/local/LocalDatabaseModule.kt]: Fixed, ty.

**Pull request merged**